server/schedule/placement/label_constraint.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package placement
+
+import (
+	"github.com/pingcap/pd/pkg/slice"
+	"github.com/pingcap/pd/server/core"
+)
+
+// LabelConstraintOp defines how a LabelConstraint matches a store. It can be one of
+// 'in', 'notIn', 'exists', or 'notExists'.
+type LabelConstraintOp string
+
+const (
+	// In restricts the store label value should in the value list.
+	In LabelConstraintOp = "in"
+	// NotIn restricts the store label value should not in the value list.
```

Comment:
Should specify the store should have the key? sometimes the target set may `NotIn`&&`NotExists`.

Reply:
I think it is reasonable that both `notIn` and `notExists` be satisfied when a store does not have the key.

Reply:
fixed.